.github/workflows/check-release.yml

```diff
@@ -36,20 +39,30 @@ jobs:
           key: ${{ runner.os }}-pip-${{ hashFiles('setup.cfg', 'setup.py') }}
           restore-keys: |
             ${{ runner.os }}-pip-
-      - name: Cache checked links
-        uses: actions/cache@v2
-        with:
-          path: ~/.cache/pytest-link-check
-          key: ${{ runner.os }}-linkcheck-${{ hashFiles('**/.md') }}-md-links
-          restore-keys: |
-            ${{ runner.os }}-linkcheck-
       - name: Upgrade packaging dependencies
         run: |
           pip install --upgrade pip setuptools wheel jupyter-packaging~=0.10 --user
       - name: Install Dependencies
         run: |
           pip install .
       - name: Check Release
+        if: ${{ matrix.group == 'check_release' }}
         uses: jupyter-server/jupyter_releaser/.github/actions/check-release@v1
         with:
-          token: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Cache checked links
+        if: ${{ matrix.group == 'link_check' }}
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pytest-link-check
+          key: ${{ runner.os }}-linkcheck-${{ hashFiles('**/.md') }}-md-links
+          restore-keys: |
+            ${{ runner.os }}-linkcheck-
+      - name: Run Link Check
+        if: ${{ matrix.group == 'link_check' }}
+        uses: jupyter-server/jupyter_releaser/.github/actions/check-links@v1
+      - name: Upload Distributions
```

Comment:
We could also have the `if: ${{ matrix.group == 'check_release' }}` condition here too for uploading the distributions as artifacts.

Reply:
Good point, I'll open another PR.